query/src/sessions/query_ctx.rs

```diff
@@ -295,37 +295,40 @@ impl QueryContext {
     }
 
     // Get the storage data accessor by config.
-    pub fn get_storage_accessor(&self) -> Result<Arc<dyn DataAccessor>> {
+    // TODO(xuanwo): we can build dal backend only once.
+    pub fn get_storage_accessor(&self) -> Result<Operator> {
         let storage_conf = &self.get_config().storage;
-        let scheme_name = &storage_conf.storage_type;
-        let scheme = StorageScheme::from_str(scheme_name)?;
-        let da: Arc<dyn DataAccessor> = match scheme {
-            StorageScheme::S3 => {
+        let schema_name = &storage_conf.storage_type;
+        let schema = DalSchema::from_str(schema_name)
+            .map_err(|e| ErrorCode::DalTransportError(e.to_string()))?;
+        let da: Arc<dyn Accessor> = match schema {
+            DalSchema::S3 => {
                 let conf = &storage_conf.s3;
-                Arc::new(S3::try_create(
-                    &conf.region,
-                    &conf.endpoint_url,
-                    &conf.bucket,
-                    &conf.access_key_id,
-                    &conf.secret_access_key,
-                    conf.enable_pod_iam_policy,
-                )?)
+                // TODO: Block on here to init the storage, we should find a better way.
```

Comment:
Is it possible to change the function get_storage_accessor to async?
We have do a lot of work try to remove block_on in the past :)

Reply:
Yes. It's the same concern @dantengsky raised too.

I plan to split `get_storage_accessor` into two functions (naming just for example):

- async `init_storage_accessor`: into storage accessor in the async way (and only once).
- sync `get_storage_accessor`: a simple function that returns an `Operator`, no extra APIs need to call.